backend/src/cache/main.go

```diff
@@ -15,28 +15,64 @@
 package main
 
 import (
+	"flag"
 	"log"
 	"net/http"
 	"path/filepath"
+
+	"github.com/kubeflow/pipelines/backend/src/cache/server"
 )
 
 const (
-	TlsDir      string = "/etc/webhook/certs"
-	TlsCertFile string = "cert.pem"
-	TlsKeyFile  string = "key.pem"
+	TLSDir      string = "/etc/webhook/certs"
+	TLSCertFile string = "cert.pem"
+	TLSKeyFile  string = "key.pem"
 )
 
 const (
-	MutateApi   string = "/mutate"
+	MutateAPI   string = "/mutate"
 	WebhookPort string = ":8443"
 )
 
+const (
+	initConnectionTimeout = "6m"
+
+	mysqlDBDriverDefault            = "mysql"
+	mysqlDBHostDefault              = "mysql"
+	mysqlDBPortDefault              = "3306"
+	mysqlDBGroupConcatMaxLenDefault = "4194304"
+)
+
+type WhSvrDBParameters struct {
+	dbDriver            string
+	dbHost              string
+	dbPort              string
+	dbUser              string
+	dbPwd               string
+	dbGroupConcatMaxLen string
+	namespaceToWatch    string
+}
+
 func main() {
-	certPath := filepath.Join(TlsDir, TlsCertFile)
-	keyPath := filepath.Join(TlsDir, TlsKeyFile)
+	var params WhSvrDBParameters
+	flag.StringVar(&params.dbDriver, "db_driver", mysqlDBDriverDefault, "Database driver name, mysql is the default value")
+	flag.StringVar(&params.dbHost, "db_host", mysqlDBHostDefault, "Database host name.")
+	flag.StringVar(&params.dbPort, "db_port", mysqlDBPortDefault, "Database port number.")
+	flag.StringVar(&params.dbUser, "db_user", "root", "Database user name.")
+	flag.StringVar(&params.dbPwd, "db_password", "", "Database password.")
```

Comment:
In next PR we should move to getting this from a configmap or secret.

Reply:
The configmap is in the same PR here: https://github.com/kubeflow/pipelines/pull/3266/files#diff-22261554aaee8ffbc4ba68a2ca4c4fd8